allure-generator/src/test/java/io/qameta/allure/owner/OwnerAddressParserTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright 2016-2024 Qameta Software Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.qameta.allure.owner;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class OwnerAddressParserTest {
+    @Test
+    void shouldReturnNullForNullInput() {
+        assertNull(OwnerAddressParser.parseAddress(null));
+    }
+
+    @Test
+    void shouldReturnNullForEmptyInput() {
+        assertNull(OwnerAddressParser.parseAddress(""));
+    }
+
```

Comment:
Tests look good.
For better test coverage it would be good to add a test for invalid URL and invalid email.

Reply:
@A1exKH, I added 

```kotlin
    implementation("commons-validator:commons-validator:1.7")
```

to Allure Generator `build.kts` to make it more resilient when it comes to validating URLs and emails. The remaining regexp for angle brackets parsing has been made a bit stricter too.

Please review the updated tests and implementation again. 🙏